azure/pipelines/deploy/container-app.yml

```diff
@@ -0,0 +1,149 @@
+- name: containerJsonPath
+  type: string
+  displayName: 'Path to the container.json file.'
+- name: label
+  type: string
+  displayName: 'Optional label to prefix steps in pipeline.'
+  default:
+- name: preDeploySteps
+  type: stepList
+  default: []
+  displayName: 'Optional steps to execute before deploying container app.'
+- name: postDeploySteps
+  type: stepList
+  default: []
+  displayName: 'Optional steps to execute after deploying container app.'
+
+steps:
+- task: PowerShell@2
+  displayName: '${{ coalesce(parameters.label, parameters.module) }} - Check preconditions'
+  inputs:
+    targetType: 'inline'
+    script: |
+      If (-Not (Test-Path '${{ parameters.containerJsonPath }}' -PathType Leaf)) {
+          Write-Host "'${{ parameters.containerJsonPath }}' file doesn't exist."
+          Write-Host "##vso[task.setVariable variable=skipContainerApp]true"
+      }
+      Else {
+          Write-Host "Preconditions have been met to continue container app deployment."
+          Write-Host "##vso[task.setVariable variable=skipContainerApp]false"
+      }
+
+- ${{ if ne(variables['skipContainerApp'], 'true') }}:
+  - ${{ each step in parameters.preDeploySteps }}:
+    - ${{ step }} 
+ 
+- task: AzureCLI@2
+  displayName: '${{ coalesce(parameters.label, parameters.module) }} - Pin image'
+  inputs:
+    azureSubscription: ${{ parameters.azureRegistrySubscription }} # Container registry is typically only available in production subscription.
+    scriptType: 'pscore'
+    scriptLocation: 'inlineScript'
+    failOnStandardError: true
+    inlineScript: |
+      $container = Get-Content "${{ parameters.containerJsonPath }}" | ConvertFrom-Json
+      $registry = $($container.registry -replace '.azurecr.io$')
+      $repository = $($container.repository)
+      $image = "$($repository):$($container.version)"
+           
+      # Tag image with a environment specific name so we keep track of the image that is deployed.
+      $pinnedImage = "$image-${{ parameters.environment }}"
+      Write-Host "##vso[task.setvariable variable=pinnedImage]$pinnedImage"
+     
+      Write-Host "Tagging $image as $pinnedImage"
+      az acr import --name $registry --source "$($container.registry)/$image" --image $pinnedImage --force
+
+      # Disallow deployed image modification/deletion so that it will not automatically cleaned up by purge tasks.
+      # In other words, lock the image.
+      Write-Host "Disallowing delete of $pinnedImage"
+      az acr repository update --name $registry --image $pinnedImage --write-enabled false --delete-enabled false > $null
+      
+      # Gather all tags that are relevant to the deployed environment.
+      # This means all images/tags that are currently deployed and the one we're about to deploy.
+      # Currently deployed images should be available in environment variable.
+      $currentImages = "$(${{ parameters.variablePrefix }}${{ parameters.module }}RevisionImages)"
```

Comment:
Who/what is responsible for ensuring this variable exists and it has the expected content? Is it default Azure or a custom implementation?